pkg/oc/cli/admin/createbootstrapprojecttemplate/create_bootstrap_project_template.go

```diff
@@ -2,48 +2,52 @@ package createbootstrapprojecttemplate
 
 import (
 	"errors"
-	"io"
 
 	"github.com/spf13/cobra"
 
 	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/genericclioptions"
+	"k8s.io/kubernetes/pkg/printers"
 
-	templatev1 "github.com/openshift/api/template/v1"
 	"github.com/openshift/origin/pkg/project/apiserver/registry/projectrequest/delegated"
+	"github.com/openshift/origin/pkg/user/generated/internalclientset/scheme"
 )
 
 const CreateBootstrapProjectTemplateCommand = "create-bootstrap-project-template"
 
 type CreateBootstrapProjectTemplateOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+
 	Name string
+	Args []string
+
+	Printer printers.ResourcePrinter
+
+	genericclioptions.IOStreams
 }
 
-func NewCommandCreateBootstrapProjectTemplate(f kcmdutil.Factory, commandName string, fullName string, out io.Writer) *cobra.Command {
-	options := &CreateBootstrapProjectTemplateOptions{}
+func NewCreateBootstrapProjectTemplateOptions(streams genericclioptions.IOStreams) *CreateBootstrapProjectTemplateOptions {
+	return &CreateBootstrapProjectTemplateOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme),
+		Name:       delegated.DefaultTemplateName,
+		IOStreams:  streams,
+	}
+}
 
+func NewCommandCreateBootstrapProjectTemplate(f kcmdutil.Factory, commandName string, fullName string, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewCreateBootstrapProjectTemplateOptions(streams)
 	cmd := &cobra.Command{
 		Use:   commandName,
 		Short: "Create a bootstrap project template",
 		Run: func(cmd *cobra.Command, args []string) {
-			if err := options.Validate(args); err != nil {
-				kcmdutil.CheckErr(kcmdutil.UsageErrorf(cmd, err.Error()))
-			}
-
-			template, err := options.CreateBootstrapProjectTemplate()
-			if err != nil {
-				kcmdutil.CheckErr(err)
-			}
-
-			err = kcmdutil.PrintObject(cmd, template, out)
-			if err != nil {
-				kcmdutil.CheckErr(err)
-			}
+			kcmdutil.CheckErr(o.Complete(args))
+			kcmdutil.CheckErr(o.Validate())
+			kcmdutil.CheckErr(o.Run())
 		},
 	}
 
-	cmd.Flags().StringVar(&options.Name, "name", delegated.DefaultTemplateName, "The name of the template to output.")
-	kcmdutil.AddPrinterFlags(cmd)
-
+	cmd.Flags().StringVar(&o.Name, "name", o.Name, "The name of the template to output.")
+	o.PrintFlags.AddFlags(cmd)
 	// Default to JSON
 	if flag := cmd.Flags().Lookup("output"); flag != nil {
 		flag.Value.Set("json")
```

Comment:
another nit, can be addressed in followup: use the `WithDefaultOutput` method of PrintFlags

Reply:
That make sense.